src/library/scala/Option.scala

```diff
@@ -26,8 +26,19 @@ object Option {
    *
    *  @param  x the value
    *  @return   Some(value) if value != null, None if value == null
+   *  @deprecated use [[ofNullable]] instead, as its meaning
+   *              is clearer
    */
-  def apply[A](x: A): Option[A] = if (x == null) None else Some(x)
+  @deprecated("Use `ofNullable` instead", since = "2.13.0")
+  @inline def apply[A](x: A): Option[A] = ofNullable(x)
+
+  /** An Option factory which creates Some(x) if the argument is not null,
+    *  and None if it is null.
+    *
+    *  @param  a the value
+    *  @return   Some(value) if value != null, None if value == null
+    */
+  def ofNullable[A](a: A): Option[A] = if (a == null) None else Some(a)
```

Comment:
Not useful, because the Scala type checker will just infer a lub. If you call `Option.ofNullable(5)` it will infer `A = Any` so that both `A >: Null` and `Int <: A`. Best to keep the existing type signature.

Reply:
At the same time, since an `Int` cannot be `null`, it doesn't make sense to do `Option.ofNullable(5)`. Is there any way to prevent that? I admit that inferring `Any` is not a great way to warn of a useless operation.

My main concern is about code like this:

```scala
scala> Option(null: Integer)
res0: Option[Integer] = None

scala> Option((null: Integer): Int)
res1: Option[Int] = Some(0)
```